### DISCUSSION TITLE
Get resolved tool executable path

### DISCUSSION THREAD

**Discussion by ITaluone** [nuke-build/nuke#1308] [Q&A]
Hi

Is there any way to get the resolved executable path from a tool?
E.g. if you use the `[NuGetPackage("PackageName", "ExecutableName")] Tool X;` how can you get the `AbsolutePath` of this resolved tool?

**Comment by feher** [comment]
I agree this would be needed (or something similar).

Currently, we have trouble with `LocalPath` `Tool`. If the given path is broken (e.g. the executable is not there), then we get only a null delegate. So, you just get a NullReferenceException. It'd be great to get proper API to handle this gracefully without resorting to hand-rolling our own workarounds.

**Reply by matkoch** [reply]
delegates can't provide any more data. there are several utilities like `ToolPathResolver` or `ToolPathResolver` that can help as a workaround

**Reply by feher** [reply]
OK.

I guess this is another topic but .... about the error scenarios (e.g. the tool path is broken/invalid).... I wonder if NUKE could be a bit more helpful than a null? When I tested, it didn't even show a warning (despite having an Assert.FileExists() check in ToolResolver.GetTool()).

**Reply by matkoch** [reply]
there's a `RequiredAttribute` and `Requires` target definition method
